pkg/controller/elasticsearch/certificates/transport/csr.go

```diff
@@ -113,6 +113,10 @@ func buildGeneralNames(
 }
 
 // buildCertificateCommonName returns the CN (and ES othername) entry for a given pod within a stack
-func buildCertificateCommonName(pod corev1.Pod, clusterName, namespace string) string {
-	return fmt.Sprintf("%s.node.%s.%s.es.local", pod.Name, clusterName, namespace)
+func buildCertificateCommonName(pod corev1.Pod, es esv1.Elasticsearch) string {
+	userConfiguredSuffix := es.Spec.Transport.TLS.CommonNameSuffix
+	if userConfiguredSuffix == "" {
+		return fmt.Sprintf("%s.node.%s.%s.es.local", pod.Name, es.Name, es.Namespace)
+	}
+	return fmt.Sprintf("%s.%s", pod.Name, userConfiguredSuffix)
```

Comment:
Do we want to handle user error here: e.g. `.my.suffix` vs `my.suffix` and try to do "the right thing"?

Reply:
Hmm good question. I think it's best to not do any magic here (if you manipulate this field you probably know what you're doing).

Reply:
I initially thought we would not even set the `.` in between by default.
But I don't know if there'd be any case where we wouldn't want a `.` there.